packages/surveys/src/utils/results-availability/has-user-responded.ts

```diff
@@ -0,0 +1,25 @@
+import { queryFeatures, IQueryResponse } from "@esri/arcgis-rest-feature-layer";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { IStakeholderItem } from "../../types";
+
+/**
+ * check is provided user has any survey submissions
+ * @param {IStakeholderItem.url} _url feature service url
+ * @param {string} username the username to check for survey responses
+ * @param {IRequestOptions} requestOptions The request options
+ * @returns {Promise<boolean>}
+ */
+export const hasUserResponded = (
+  _url: IStakeholderItem["url"],
```

Comment:
I'm curious how this parameter works. It looks optional with type defined by the interface attribute type?

Reply:
This is probably an overreach on my part learning TS. My guess is that it resolves to `_url: string`, since the definition of `IStakeholderItem > IItem.url` is `string`. I should probably just make it `string` for clarity

Reply:
why is it `_url` instead of `url`?

Reply:
> why is it `_url` instead of `url`?

just so i can use `url` in the function itself